cmd/opentelemetry-collector/go.mod

```diff
@@ -0,0 +1,9 @@
+module github.com/jaegertracing/jaeger/cmd/opentelemetry-collector
```

Comment:
@yurishkuro this is an example fo go submodule. 

By default Jaeger version would be resolved from OTEL collector, but with `replace` we are able to point it to the current master/head. If we do a breaking change we can pin the version to a version prior the change and collaborate with OTEL to update the code there.